src/psyclone/transformations.py

```diff
@@ -1361,3 +1363,83 @@ def apply(self, loop, depth=None):
         loop.update_halo_exchanges()
 
         return schedule, keep
+
+
+class LoopSwapTrans(Transformation):
+    ''' Provides a loop-swap transformation.
+        For example:
+
+        >>> from parse import parse
+        >>> from psyGen import PSyFactory
+        >>> ast,invokeInfo=parse("dynamo.F90")
+        >>> psy=PSyFactory("dynamo0.1").create(invokeInfo)
+        >>> schedule=psy.invokes.get('invoke_v3_kernel_type').schedule
+        >>> schedule.view()
+        >>>
+        >>> from transformations import LoopSwapTrans
+        >>> swap=LoopSwapTrans()
+        >>> new_schedule,memento=swap.apply(schedule.children[0],
+                                             schedule.children[1])
+        >>> new_schedule.view()
+    '''
+
+    def __str__(self):
+        return "Exchange the order of two nested loops: inner becomes " + \
+               "outer and vice versa"
+    @property
+    def name(selfself):
+        '''Returns the name of this transformation as a string'''
+        return "LoopSwap"
+
+    def _validate(self, node_outer):  # pylint: disable=no-self-use
+        ''' validity checks for input arguments '''
+
+        from .psyGen import Loop
+        if not isinstance(node_outer, Loop):
+            raise TransformationError("Error in LoopSwap transformation. "
+                                      "Given node is not a loop.")
+
+        node_inner = node_outer.children[0]
+        # Check that the supplied Node is a Loop
+        if not isinstance(node_inner, Loop):
+            raise TransformationError("Error in LoopSwap transformation. "
+                                      "First inner statement is not a loop.")
+
+        if len(node_outer.children) != 1:
+            raise TransformationError("Error in LoopSwap transformation. "
+                                      "More than one statement in outer loop.")
+
+    def apply(self, outer):  # pylint: disable=arguments-differ
+        ''' Swaps the loop by :py:obj:`inner`with its inner loop. '''
```

Comment:
Please correct this docstring and add documentation for the interface to this routine.

Reply:
Done, but I am not sure how to properly specify rtype a tuple of a schedule and momento.